backend/app/app.go

```diff
@@ -0,0 +1,121 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"kubecloud/internal"
+	"kubecloud/middlewares"
+	"kubecloud/models/sqlite"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+)
+
+// App holds all configurations for the app
+type App struct {
+	router     *gin.Engine
+	httpServer *http.Server
+	config     internal.Configuration
+	handlers   Handler
+}
+
+// NewApp create new instance of the app with all configs
+func NewApp(config internal.Configuration) (*App, error) {
+	router := gin.Default()
+
+	tokenHandler := internal.NewTokenHandler(
+		config.JWT.Secret,
+		time.Duration(config.JWT.AccessTokenExpiryMinutes)*time.Minute,
+		time.Duration(config.JWT.RefreshTokenExpiryHours)*time.Hour,
+	)
+
+	db, err := sqlite.NewSqliteStorage(config.Database.File)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to create user storage")
+		return nil, fmt.Errorf("failed to create user storage: %w", err)
+	}
+
+	handler := NewHandler(tokenHandler, db, config)
+
+	app := &App{
+		router:   router,
+		config:   config,
+		handlers: *handler,
+	}
+
+	app.registerHandlers()
+
+	return app, nil
+
+}
+
+// registerHandlers registers all routes
+func (app *App) registerHandlers() {
+	v1 := app.router.Group("/api/v1")
+	{
+		v1.POST("/register", app.handlers.RegisterHandler)
+		v1.POST("/login", app.handlers.LoginUserHandler)
+		v1.POST("/refresh", app.handlers.RefreshTokenHandler)
```

Comment:
refresh needs to be authorized

Reply:
- how ? i will send the expired access token to it so it won't even pass the middlware.